roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -226,6 +226,67 @@ object tuple {
     }
   }
 
+  /**
+    * Define the tuple-new primitive.
+    * This returns a new Tuple containing the specified n Obs
+    * e.g. (new Tuple 1 2 3 4 5 6) ==> [1 2 3 4 5 6]
+    *
+    * WARNING: this primitive (like tplNewN, tplexprNew and tplexprNewN)
+    * takes an extra argument in the leading slot, allowing these
+    * primitives to be bound directly to operations.  That is, the
+    * operation new is bound to tplNew in a prototypical tuple Tuple,
+    * then the expression
+    *
+    * 	(new Tuple a b c)
+    *
+    * will be equivalent to
+    *
+    * 	(tplNew Tuple a b c)
+    *
+    * Since we want to produce [a b c], we need to ignore the first arg.
+    */
+  object tplNew extends Prim {
+    override val name: String = "tuple-new"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = MaxArgs
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
+      val elem  = ctxt.argvec.elem
+      val nargs = ctxt.nargs
+
+      Right(Tuple(elem.slice(1, nargs - 1)))
+    }
+  }
+
+  /**
+    * Define the tuple-new-n primitive.
+    * This returns a new Tuple containing the specified n duplicated Obs
+    *
+    * See warning in tplNew about the rationale for ignoring ARG(0).
+    * 	(tplNewN dummy n init)
+    *
+    * e.g.(tplNewN [] 3 'a) ===>	['a 'a 'a]
```

Comment:
This seems to be an error in the original comments.
I guess we should fix it to: `(tuple-new-n [] 3 'a)`.

Reply:
I agree.  Done.